shipengine_sdk/http_client/client.py

```diff
@@ -13,12 +14,48 @@
 from shipengine_sdk import __version__
 
 from ..errors import ShipEngineError
+from ..events import (
+    EventOptions,
+    RequestSentEvent,
+    ResponseReceivedEvent,
+    ShipEngineEvent,
+    emit_event,
+)
 from ..jsonrpc.process_request import handle_response, wrap_request
 from ..models import ErrorCode, ErrorSource, ErrorType
 from ..shipengine_config import ShipEngineConfig
 from ..util.sdk_assertions import is_response_404, is_response_429, is_response_500
 
 
+def base_url(config) -> str:
+    return config.base_uri if os.getenv("CLIENT_BASE_URI") is None else os.getenv("CLIENT_BASE_URI")
+
+
+def check_for_errors(status_code: int, response_body: Dict[str, Any], config) -> None:
+    is_response_404(status_code=status_code, response_body=response_body, config=config)
+    is_response_429(status_code=status_code, response_body=response_body, config=config)
+    is_response_500(status_code=status_code, response_body=response_body)
```

Comment:
I might follow up sometime on this out of curiosity, but I think everything looks great overall and read and flowed nicely.

Reply:
Yes, I am happy to sync on this anytime! The `is_response_123` methods were making the client class body enormous so I tried to move things out into functions where I would help that cause. I would ultimately like to have one function `check_response_for_error()` that has all the `is_response_123` logic int

Reply:
Actually, I might make that change before closing/merging this PR.

Reply:
Change made! Thanks Nate.

Reply:
Change made! Thanks Nate.